pkg/handlers/generic/mutation/taints/inject_worker.go

```diff
@@ -103,25 +109,40 @@ func (h *taintsWorkerPatchHandler) Mutate(
 
 	if err := patches.MutateIfApplicable(
 		obj, vars, &holderRef,
-		selectors.WorkersConfigTemplateSelector(eksbootstrapv1.GroupVersion.String(), "EKSConfigTemplate"), log,
-		func(obj *eksbootstrapv1.EKSConfigTemplate) error {
+		selectors.WorkersConfigTemplateSelector(eksbootstrapv1.GroupVersion.String(), "NodeadmConfigTemplate"), log,
+		func(obj *eksbootstrapv1.NodeadmConfigTemplate) error {
 			log.WithValues(
 				"patchedObjectKind", obj.GetObjectKind().GroupVersionKind().String(),
 				"patchedObjectName", ctrlclient.ObjectKeyFromObject(obj),
-			).Info("adding taints to worker node EKS config template")
-			if obj.Spec.Template.Spec.KubeletExtraArgs == nil {
-				obj.Spec.Template.Spec.KubeletExtraArgs = make(map[string]string, 1)
+			).Info("adding taints to worker NodeadmConfig template")
+			kubeletOptions := obj.Spec.Template.Spec.Kubelet
+			var runtimeConfigFromNodeadm *runtime.RawExtension
+			var flags []string
+			newTaints := toEKSConfigTaints(taintsVar)
+			hasRegisterTaintsFlag := false
+			if kubeletOptions != nil {
+				runtimeConfigFromNodeadm = kubeletOptions.Config
+				for _, flag := range kubeletOptions.Flags {
+					if strings.HasPrefix(flag, "--register-with-taints=") {
+						hasRegisterTaintsFlag = true
+						existingTaints := strings.Split(flag, "--register-with-taints=")
+						if len(existingTaints) != 2 {
+							return fmt.Errorf("expected flag register-with-taints to be able to split got %v", existingTaints)
+						}
+						taintsFromFlags := existingTaints[1]
+						flags = append(flags, fmt.Sprintf("--register-with-taints=%s,%s", taintsFromFlags, newTaints))
+						continue
+					}
+					flags = append(flags, flag)
+				}
```

Comment:
Can we use flag parsing logic like https://play.golang.com/p/jXtiWTdEK8C instead? This will handle getting the values if the `--register-with-taints` flag is specified with/without `=` to specify values and also if specified multiple times.

Reply:
if flags can be issued multiple times why do we need to parse this at all?

Reply:
that's because of the way kubeadm collects those flags in a map and adds them to the kubelet invocation. what about if you run `kubelet` binary with it specified multiple times?